File: BackUp/2014/version-2014-01-09/AttackoftheRGB/Assets/Scripts/C#/healthBarScript.cs
using UnityEngine;
using System.Collections;

public class healthBarScript : MonoBehaviour 
{
	public Texture2D healthBar;
	private double healthMax = 1000.0;
	public double health = 1000.0;
	public bool repair = false;
	
	private void Start()
	{
		health = 1000.0;
		transform.localScale = new Vector3(0.2f,0,1);
	}
	
	private void OnTriggerExit(Collider hit)
	{
	}
	
	private void Update()
	{
		//checks whether the health bar is at a length where it is visible
		if(repair == true)
		{
			if(health > 0 && health < 1000)
			{
				//print(health);
				health += 500 * Time.deltaTime;
				print (500 * Time.deltaTime);
				//transform.localScale += new Vector3(0.2f,0.0f,0.0f);
			}
			else
			{
				//print("Your Dead");
				
			}
						
			if(health > 1000)
			{
				health = healthMax;
				transform.localScale = new Vector3(0.2f,0,1);	//change scale of healthbar					
			}
			
			if(transform.localScale.x >= 0.2)
			{
				transform.localScale = new Vector3(0.2f,0,1);
				health = 1000;
			}
			
			if(health < 0)
			{
				health = 0;
				transform.localScale =  new Vector3(0,0,1);
				//you're dead
			}
			
			if(transform.localScale.x < 0)
			{
				transform.localScale =  new Vector3(0,0,1);
				health = 0;
				//you're dead;
			}
		}
	}
}

File: AttackoftheRGB/Assets/Standard Assets/JavaScript/HealthBarScript.js
#pragma strict

var healthBar : Texture2D; //healthBar GUI
private var healthmax = 1000f; //the reason why the number is this big - is due to how long it take to get from the 1st shelter toe the ship
public var health = 100f;

var repair = true;

function Start () 
{
	health = 100f;
	transform.localScale.x = 0.1f;
}

function OnTriggerEnter(hit : Collider)
{
	if(hit.gameObject.tag == "ally")
	{
		//print("You have entered a shelter. health will cease to depleat now"); //check if player has collided with a shelter - health will not depleat.
		//healthBar.
		//health + 500;		
	}	
}
function Update () 
{
	
	//checks whether the health bar is at a length where it is visible
	if(repair == true)
	{
		if(health > 0)
		{
			//print(health);
			health += 1.75 * Time.deltaTime;
			//print("bar length " + transform.localScale.x); //check players health.
			transform.localScale += new Vector3(health / 60000000f,0,transform.localScale.z);	//change scale of healthbar
			//t//ransform.position.x -= 0.00075;
		}
		else
		{
			//print("Your Dead");
			
		}
		
		
		if(health > 1000)
		{
			health = healthmax;
			transform.localScale.x = 0.18;	//change scale of healthbar		
			
		}
		if(transform.localScale.x >= 0.18)
		{
			transform.localScale.x = 0.18;
			health = 1000;
		}
		if(health < 0)
		{
			health = 0;
			transform.localScale.x = 0;
			//your dead
		}
		if(transform.localScale.x < 0)
		{
			transform.localScale.x = 0;
			health = 0;
			//yourdead;
		}
	}
	
	

}
